Add a description of project tiers (#152)
maplibre/maplibre: PROJECT_TIERS.md

# Project Tiers

The MapLibre Organization maintains the map rendering frameworks MapLibre GL JS and MapLibre GL Native and fosters an ecosystem of plugins and tools around these libraries to produce the best maps in the world.

> The three project tiers Core, Supported, and Hosted describe qualitatively how donated resources are distributed among the projects.

Reviewer: ```suggestion
The three project tiers Core, Supported, and Hosted describe qualitatively how organization resources are distributed among the projects.
```

Reviewer: I'd not specify "donated" since the resources may or may not be donated explicitly.

## Tiers

### Core Project

> A project that the MapLibre Organization is committed to in a long-term way. A substantial amount of funding is spent on the project in the form of paid Maintainer time and paid Bounties to ensure its long-term success.

Reviewer: ```suggestion
A project that the MapLibre Organization is committed to long-term. A substantial amount of resources is allocated to the project in the form of paid Maintainer time and paid Bounties to ensure its long-term success.
```

### Supported Project

> A project that the MapLibre Organization can allocate funding to in the form of paid Bounties. Supported Projects receive a substantially smaller amount of funding than Core Projects and the MapLibre Organization cannot guarantee any level of long-term support.

Reviewer: ```suggestion
A project that the MapLibre Organization can allocate resources to in the form of paid Bounties. Supported Projects receive a substantially smaller amount of resources than Core Projects and the MapLibre Organization does not guarantee any level of long-term support.
```

### Hosted Project

> A project that the MapLibre Organization does not allocate any funding to. The project is hosted in the MapLibre GitHub organization to facilitate collaboration between the people who are interested in the project.

Reviewer: ```suggestion
A project that the MapLibre Organization does not allocate any resources to. The project is hosted in the MapLibre GitHub organization to facilitate collaboration between the people who are interested in the project.
```

# Project Tiers

The MapLibre Organization maintains the map rendering frameworks MapLibre GL JS and MapLibre GL Native and fosters an ecosystem of plugins and tools around these libraries to produce the best maps in the world.

The three project tiers Core, Supported, and Hosted describe qualitatively how donated resources are distributed among the projects.

## Tiers

### Core Project

A project that the MapLibre Organization is committed to in a long-term way. A substantial amount of funding is spent on the project in the form of paid Maintainer time and paid Bounties to ensure its long-term success.

### Supported Project

> A project that the MapLibre Organization can allocate funding to in the form of paid Maintainer time and paid Bounties. Supported Projects receive a substantially smaller amount of funding than Core Projects and the MapLibre Organization cannot guarantee any level of long-term support.

Reviewer 2: Only bounties?

Author: Reduced to Bounties now. Let us keep in mind though that Bounties will need some management work which someone will need to do as a volunteer.

### Hosted Project

A project that the MapLibre Organization does not allocate any funding to. The project is hosted in the MapLibre GitHub organization to facilitate collaboration between the people who are interested in the project.

## Process

New projects start in the Hosted tier.

If you would like to propose to change the tier of a project, please edit this file in a pull request.

Changing the tier of a project requires approval by the MapLibre Governing Board, as the Board is responsible for all financial decisions.

## List of Projects

### Core

* [maplibre-gl-js](https://github.com/maplibre/maplibre-gl-js)
* [maplibre-gl-native](https://github.com/maplibre/maplibre-gl-native)

### Supported

* Currently none.

### Hosted

> * [maplibre-gl-native-boost](https://github.com/maplibre/maplibre-gl-native-boost)

Author: Good point, fixed that

* [mbtileserver-rs](https://github.com/maplibre/mbtileserver-rs)
* [maplibre-plugins-android](https://github.com/maplibre/maplibre-plugins-android)
* [demotiles](https://github.com/maplibre/demotiles)
* [maplibre-gl-native-distribution](https://github.com/maplibre/maplibre-gl-native-distribution)
* [maplibre-gl-directions](https://github.com/maplibre/maplibre-gl-directions)
* [martin-landing-page](https://github.com/maplibre/martin-landing-page)

> * [maplibre-native-base](https://github.com/maplibre/maplibre-native-base)

Reviewer 3: This is a part of the maplibre-gl-native C++ core and should be just as much in core as maplibre-gl-native. It'll likely be merged into the main repo at some point.

* [maplibre.github.io](https://github.com/maplibre/maplibre.github.io)
* [awesome-maplibre](https://github.com/maplibre/awesome-maplibre)
* [qt-geoservices-maplibre-gl](https://github.com/maplibre/qt-geoservices-maplibre-gl)
* [font-maker](https://github.com/maplibre/font-maker)
* [maplibre-react-native](https://github.com/maplibre/maplibre-react-native)
* [maplibre-gl-leaflet](https://github.com/maplibre/maplibre-gl-leaflet)
* [maplibre-navigation-android](https://github.com/maplibre/maplibre-navigation-android)
* [maplibre-gl-geocoder](https://github.com/maplibre/maplibre-gl-geocoder)

> * [maplibre-gl-js-docs](https://github.com/maplibre/maplibre-gl-js-docs)

Reviewer 3: Our maplibre-gl-js docs should also be eligible to receive maintainer time.

Reviewer 2: I agree with what @birkskyum said.
It might be that the js docs will be part of maplibre-gl-js in the future, but even if not, it requires the maintainer's time.

Author: I agree as well. Added maplibre-gl-js-docs to the core projects.

Author: I agree as well. Added maplibre-gl-js-docs to the core projects.
